rosidl_typesupport_cpp/resource/srv__type_support.cpp.em

```diff
@@ -109,10 +120,70 @@ static const type_support_map_t _@(service.namespaced_type.name)_service_typesup
   &_@(service.namespaced_type.name)_service_typesupport_data.data[0],
 };
 
+@{event_type = '::'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]) + SERVICE_EVENT_MESSAGE_SUFFIX}
+
+void *
+rosidl_@('_'.join([package_name, *interface_path.parents[0].parts, service.namespaced_type.name]))_event_message_create(
+  const rosidl_service_introspection_info_t * info,
+  rcutils_allocator_t * allocator,
+  const void * request_message,
+  const void * response_message,
+  bool enable_message_payload)
+{
+  if (nullptr == info) {
+    throw std::invalid_argument("service introspection info struct cannot be null");
+  }
+  if (nullptr == allocator) {
+    throw std::invalid_argument("allocator cannot be null");
+  }
+  auto * event_msg = static_cast<@event_type *>(allocator->allocate(sizeof(@event_type), allocator->state));
+  if (nullptr == event_msg) {
+    throw std::invalid_argument("allocation failed for service event message");
+  }
+  event_msg = new(event_msg) @(event_type)();
+
+  event_msg->info.set__event_type(info->event_type);
+  event_msg->info.set__sequence_number(info->sequence_number);
+  event_msg->info.stamp.set__sec(info->stamp_sec);
+  event_msg->info.stamp.set__nanosec(info->stamp_nanosec);
+
+  std::array<uint8_t, 16> client_id;
+  std::move(std::begin(info->client_id), std::end(info->client_id), client_id.begin());
+  event_msg->info.client_id.set__uuid(client_id);
+
+  // TODO(jacobperron): consider removing this argument and let users pass nullptr for both request and response messages
+  if (!enable_message_payload) {
+    return event_msg;
+  }
```

Comment:
Looking for feedback on this TODO

Reply:
I think removing the `bool` makes sense, I'm going to implement that change